Incompatibility with mtl >=2.3

logic-TPTP  0.5.0.0 and `master` (1c9a28612786e92e097310b5bd334b49b0fe5994) failed to compile with `mtl-2.3.1` with the following errors:

```
/Users/sakai/src/logic-TPTP/Codec/TPTP/Base.hs:364:69: error:
    Variable not in scope:
[ 6 of 11] Compiling Lexer
      ap
        :: Gen ([AtomicWord] -> TPTP_Input_ c0)
           -> Gen [a0] -> Gen TPTP_Input
    Suggested fix:
      Perhaps use one of these:
        data constructor ‘Ap’ (imported from Data.Monoid),
        ‘map’ (imported from Data.Set), ‘map’ (imported from Prelude)
    |
364 |                                   , (1, Include `fmap` arbLowerWord `ap`
    |                                                                     ^^^^

/Users/sakai/src/logic-TPTP/Codec/TPTP/Base.hs:377:60: error:
    Variable not in scope:
      ap :: Gen (UsefulInfo -> Annotations) -> Gen a1 -> Gen Annotations
    Suggested fix:
      Perhaps use one of these:
        data constructor ‘Ap’ (imported from Data.Monoid),
        ‘map’ (imported from Data.Set), ‘map’ (imported from Prelude)
    |
377 |                             , Annotations `fmap` arbitrary `ap` arbitrary
    |                                                            ^^^^

[11 of 11] Compiling Util
/Users/sakai/src/logic-TPTP/Codec/TPTP/Base.hs:421:38: error:
    Variable not in scope:
      liftM2 :: (a2 -> [a2] -> [a2]) -> Gen a3 -> Gen [a4] -> Gen [V]
    Suggested fix:
      Perhaps use one of these:
        ‘liftA2’ (imported from Control.Applicative),
        ‘lift’ (imported from Control.Monad.State),
        ‘liftA’ (imported from Control.Applicative)
    |
421 |                                x2 <- liftM2 (:) arbitrary (argsFreq (\nargs -> vectorOf nargs arbitrary))
    |                                      ^^^^^^

/Users/sakai/src/logic-TPTP/Codec/TPTP/Base.hs:454:82: error:
    Variable not in scope:
      ap :: Gen ([term0] -> Term0 term0) -> m0 [a5] -> Gen (Term0 term)
    Suggested fix:
      Perhaps use one of these:
        data constructor ‘Ap’ (imported from Data.Monoid),
        ‘map’ (imported from Data.Set), ‘map’ (imported from Prelude)
    |
454 |             go 0 = frequency [ (2,Var <$> arbitrary), (1,FunApp `fmap` arbitrary `ap` return[] ) ]
    |                                                                                  ^^^^
```

```
/Users/sakai/src/logic-TPTP/Codec/TPTP/Diff.hs:278:22: error:
    Variable not in scope:
      liftM2
        :: (F c0 -> F c0 -> F c0)
           -> Gen Formula -> Gen Formula -> Gen (F c)
    |
278 |                      liftM2 (.&.) (resize ileft diffGenF) (resize (i-1-ileft) diffGenF)
    |                      ^^^^^^
[ 7 of 11] Compiling Parser

/Users/sakai/src/toysolver/logic-TPTP/Codec/TPTP/Diff.hs:279:52: error:
    Variable not in scope:
      ap :: Gen (T Identity -> F Identity) -> Gen Term -> Gen (F c)
    Suggested fix:
      Perhaps use one of these:
        ‘a’ (line 279), ‘map’ (imported from Prelude)
    |
279 |                   , let a=diffGenT in fmap (.=.) a `ap` a
    |                                                    ^^^^
```

This is because mtl-2.3 remove re-exports of `Control.Monad`, `Control.Monad.Fix` and `Data.Monoid` modules.
https://hackage.haskell.org/package/mtl-2.3.1/changelog

--- a/Codec/TPTP/Base.hs
+++ b/Codec/TPTP/Base.hs
@@ -17,6 +17,7 @@ module Codec.TPTP.Base where
 
 import Codec.TPTP.QuickCheck
 import Control.Applicative
+import Control.Monad
 import Control.Monad.Identity
 import Control.Monad.State
 import Data.Data

--- a/Codec/TPTP/Diff.hs
+++ b/Codec/TPTP/Diff.hs
@@ -20,6 +20,7 @@ import Test.QuickCheck hiding((.&.))
 import Codec.TPTP.Base
 import Codec.TPTP.Pretty
 import Text.PrettyPrint.ANSI.Leijen
+import Control.Monad
 import Control.Monad.Identity
 
 

